### PR TITLE
Adds explosion when attempting to microwave metal / bugfix

### PR DESCRIPTION
**Pull request by Tin-Man-Tim**
## About the PR
Adds a small explosion for when users attempt to microwave metal. There was also the issue of the microwave not ejecting its contents on destruction from microwaving metal, which should also be fixed in this pr.

old functionality:
https://github.com/space-wizards/space-station-14/assets/73014819/c58b31e0-0190-467b-8eb3-76730029cabd

new functionality: 
https://github.com/space-wizards/space-station-14/assets/73014819/7f732b01-17e1-45b5-bb69-13b1dd26d96d

**Changelog**

:cl:
- tweak: Attempting to microwave metal now can cause the microwave to spark and explode.

### CODE REVIEW

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 1]
```diff
@@ -353,7 +356,7 @@ public static bool HasContents(MicrowaveComponent component)
         /// It does not make a "wzhzhzh" sound, it makes a "mmmmmmmm" sound!
         /// -emo
         /// </remarks>
-        public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user)
+        public void StartCooking(EntityUid uid, MicrowaveComponent component, EntityUid? user)
```
Comment (deathride58) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: ?

Comment (GreyMario) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: wzhzhzh perfectly describes the spaghetti that drives the microwave though

Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: pretty much

Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: > ?

If the change isn't warranted then I can revert it, lmk.

Comment (EmoGarbage404) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: you should revert it. it's hard to explain, but this function header must rename as it was.

Comment (deltanedas) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:359]: unacceptable breaking change to microwave api

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 2]
```diff
@@ -377,9 +380,15 @@ public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user
                 // destroy microwave
                 if (_tag.HasTag(item, "MicrowaveMachineUnsafe") || _tag.HasTag(item, "Metal"))
                 {
-                    component.Broken = true;
-                    SetAppearance(uid, MicrowaveVisualState.Broken, component);
-                    _audio.PlayPvs(component.ItemBreakSound, uid);
+                    if (_tag.HasTag(item, "Metal"))
+                    {
+                        _explosionSystem.QueueExplosion(uid, "Default", 10, 0.5f, 1, canCreateVacuum: false);
```
Comment (EmoGarbage404) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:385]: you can just add the ExplosiveComponent to the microwave and then call Explode or w/e. That way you don't hardcode these values and other microwaves in the future can adjust them.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by VektorZ1** [comment]
Does this apply to forks and spoons? :trollface:

**Comment by Tin-Man-Tim** [comment]
> Does this apply to forks and spoons? :trollface:

yes.

**Comment by Tin-Man-Tim** [comment]
https://github.com/space-wizards/space-station-14/assets/73014819/5d65d06c-925f-4195-a639-0219d0ac4ec8

**Comment by Tin-Man-Tim** [comment]
any major code smell?

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 2 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: > you can just add the ExplosiveComponent to the microwave and then call Explode or w/e. That way you don't hardcode these values and other microwaves in the future can adjust them.

Did I resolve it correctly?

File: Content.Server/Kitchen/Components/ActiveMicrowaveComponent.cs [hunk 3]
```diff
@@ -11,6 +11,9 @@ public sealed partial class ActiveMicrowaveComponent : Component
     [ViewVariables(VVAccess.ReadWrite)]
     public float CookTimeRemaining;
 
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float ElapsedTime;
```
Comment (moonheart08) [inline Content.Server/Kitchen/Components/ActiveMicrowaveComponent.cs:15]: Generally standard to instead store the TimeSpan at which you started counting, and calculate the elapsed using IGameTiming's curtime.

File: Content.Server/Kitchen/Components/MicrowaveComponent.cs [hunk 4]
```diff
@@ -73,6 +73,19 @@ public sealed partial class MicrowaveComponent : Component
 
         [DataField, ViewVariables(VVAccess.ReadWrite)]
         public int Capacity = 10;
+
+        /// <summary>
+        ///     cooldown interval for how often the microwave can sustain damage while cooking metal.
```
Comment (moonheart08) [inline Content.Server/Kitchen/Components/MicrowaveComponent.cs:78]: grammar/capitalization (nitpick)

File: Content.Server/Kitchen/Components/MicrowaveComponent.cs [hunk 5]
```diff
@@ -73,6 +73,19 @@ public sealed partial class MicrowaveComponent : Component
...
+        /// </summary>
+        [DataField("destructionCooldown"), ViewVariables(VVAccess.ReadWrite)]
+        public float DestructionCooldown = 1.0f;
+
+        /// <summary>
+        ///     if the microwave currently contains a metalic object or item that is unsafe for microwave use.
```
Comment (moonheart08) [inline Content.Server/Kitchen/Components/MicrowaveComponent.cs:84]: grammar/capitalization (nitpick)

File: Content.Server/Kitchen/Components/MicrowaveComponent.cs [hunk 6]
```diff
@@ -73,6 +73,19 @@ public sealed partial class MicrowaveComponent : Component
...
+        /// </summary>
+        [DataField("containsMetal"), ViewVariables(VVAccess.ReadWrite)]
+        public bool ContainsMetal = false;
```
Comment (moonheart08) [inline Content.Server/Kitchen/Components/MicrowaveComponent.cs:87]: I feel like this property has implied behavior, it should be named after the behavior it causes, not what causes it to be set.

File: Content.Server/Kitchen/Components/ActiveMicrowaveComponent.cs (hunk 3 above)
Comment (moonheart08) [reply Content.Server/Kitchen/Components/ActiveMicrowaveComponent.cs:15]: Additionally make sure to use a timespan serializer, as to not break saving active microwaves. (which might be broken anyway who knows but best not make it worse)

File: Content.Server/Lightning/LightningSystem.cs [hunk 7]
```diff
@@ -53,10 +54,14 @@ public void ShootLightning(EntityUid user, EntityUid target, string lightningPro
         var spriteState = LightningRandomizer();
         _beam.TryCreateBeam(user, target, lightningPrototype, spriteState);
 
-        var ev = new HitByLightningEvent(user, target);
-        RaiseLocalEvent(target, ref ev);
+        if (lightningPrototype != "Spark") // we don't want certain prototypes to trigger lightning level events
```
Comment (moonheart08) [inline Content.Server/Lightning/LightningSystem.cs:57]: Prototype comparisons, especially hardcoded ones, are a big "do not do" in this codebase. Not 100% on a better approach but this absolutely cannot be done.

File: Resources/Prototypes/Entities/Structures/Machines/microwave.yml [hunk 8]
```diff
@@ -83,6 +83,7 @@
       behaviors:
       - !type:DoActsBehavior
         acts: ["Breakage"]
+      - !type:ExplodeBehavior
```
Comment (moonheart08) [inline Resources/Prototypes/Entities/Structures/Machines/microwave.yml:86]: Do we really want any damage to a microwave to cause it to explode, even just someone punching it really hard?

**Review by moonheart08 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Structures/Machines/microwave.yml (hunk 8 above)
Comment (EmoGarbage404) [reply Resources/Prototypes/Entities/Structures/Machines/microwave.yml:86]: if you meant to make the microwave explosive, like i suggested in my review, you just have to add the ExplosiveComponent. The explode behavior is only for if you want it on destruction (no, imo)

**Review by ShadowCommander — CHANGES_REQUESTED** [review]
Looking good! Just needs a few formatting fixes and a small timing change.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 9]
```diff
@@ -384,6 +396,35 @@ public static bool HasContents(MicrowaveComponent component)
...
+            }
+
+            if (component.IsSparking)
+            {
+                if (_random.Prob(.1f))
+                {
+                    _explosion.TriggerExplosive(uid);
+                    return false;  // microwave is fucked, stop the cooking.
+                }
+                if (_random.Prob(.75f))
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:419]: Is there a particular reason to otherwise not always spark?

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 10]
```diff
@@ -45,12 +49,20 @@ public sealed class MicrowaveSystem : EntitySystem
...
+        [Dependency] private readonly SharedDestructibleSystem _destruction = default!;
         [Dependency] private readonly SharedContainerSystem _sharedContainer = default!;
         [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
         [Dependency] private readonly TagSystem _tag = default!;
         [Dependency] private readonly TemperatureSystem _temperature = default!;
         [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
         [Dependency] private readonly HandsSystem _handsSystem = default!;
+
+        private TimeSpan _targetTime = TimeSpan.Zero;
+
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:65]: This line should come in a block below all of the [Dependency] lines for systems.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 11]
```diff
@@ -384,6 +396,35 @@ public static bool HasContents(MicrowaveComponent component)
             return component.Storage.ContainedEntities.Any();
         }
 
+        /// <summary>
+        /// handles the attempted cooking of unsafe objects
+        /// </summary>
+        /// <remarks>
+        /// returns true if the microwave can continue to cook, false if it can't.
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:403]: Capitalization on comments.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 12]
```diff
@@ -517,10 +559,22 @@ public override void Update(float frameTime)
             base.Update(frameTime);
 
             var query = EntityQueryEnumerator<ActiveMicrowaveComponent, MicrowaveComponent>();
+
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:562]: Here the declaration of `query` above is closely associated with this while loop, so you can get rid of this extra whitespace.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 13]
```diff
@@ -412,20 +455,19 @@ public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user
...
                 {
-                    component.Broken = true;
-                    SetAppearance(uid, MicrowaveVisualState.Broken, component);
-                    _audio.PlayPvs(component.ItemBreakSound, uid);
-                    return;
+                    component.IsSparking = true;
                 }
 
-                if (_tag.HasTag(item, "MicrowaveSelfUnsafe") || _tag.HasTag(item, "Plastic"))
+                if (_tag.HasTag(item, "Plastic"))
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:463]: Why was `MicrowaveSelfUnsafe` removed?

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 14]
```diff
@@ -517,10 +559,22 @@ public override void Update(float frameTime)
...
+
                 active.CookTimeRemaining -= frameTime;
+
+                if (_gameTiming.CurTime >= _targetTime)
+                {
+                    if (!HandleUnsafeItems(uid, microwave))
+                    {
+                        _destruction.BreakEntity(uid);
+                    }
+                    _targetTime += TimeSpan.FromSeconds(1);
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:574]: Do you mean to add `continue` here to stop processing the rest of this loop?

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 15]
```diff
@@ -517,10 +559,22 @@ public override void Update(float frameTime)
...
-                //check if there's still cook time left
+
                 active.CookTimeRemaining -= frameTime;
+
+                if (_gameTiming.CurTime >= _targetTime)
+                {
+                    if (!HandleUnsafeItems(uid, microwave))
+                    {
+                        _destruction.BreakEntity(uid);
+                    }
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:573]: If you add the `continue` as suggested below, you can move this loop outside of this gameTime if statement.

File: Content.Server/Lightning/LightningSystem.cs [hunk 16]
```diff
@@ -48,15 +49,20 @@ private void OnRemove(EntityUid uid, LightningComponent component, ComponentRemo
     /// <param name="user">Where the lightning fires from</param>
     /// <param name="target">Where the lightning fires to</param>
     /// <param name="lightningPrototype">The prototype for the lightning to be created</param>
-    public void ShootLightning(EntityUid user, EntityUid target, string lightningPrototype = "Lightning")
+    /// <param name="triggerLightningEvents">if the lightnings being fired should trigger lightning events.</param>
+    public void ShootLightning(EntityUid user, EntityUid target, bool triggerLightningEvents, string lightningPrototype = "Lightning")
```
Comment (Partmedia) [inline Content.Server/Lightning/LightningSystem.cs:53]: This is technically a breaking API change. Please avoid making breaking API changes by moving `triggerLightningEvents` to the last keyword argument and provide a default that preserves existing behavior.

File: Resources/Prototypes/Entities/Structures/Machines/microwave.yml [hunk 17]
```diff
@@ -83,6 +83,13 @@
       behaviors:
       - !type:DoActsBehavior
         acts: ["Breakage"]
+  - type: Explosive
+    explosionType: Default
+    maxIntensity: 20
+    totalIntensity: 10
+    intensitySlope: 5
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Structures/Machines/microwave.yml:90]: I think this explosion is too big.

**Review by Partmedia — CHANGES_REQUESTED** [review]
Thank you for your PR. In the future, please remember to separate bug fixes from content/feature changes. In particular, breaking API changes are not allowed to be bundled with feature changes.

**Comment by Partmedia** [comment]
Please also review the [Changelog guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html#writing-an-effective-changelog) and consolidate your changelog entry into one line.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 13 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:463]:
```
                if (_tag.HasTag(item, "MicrowaveMachineUnsafe") || _tag.HasTag(item, "Metal"))
                {
                    component.IsSparking = true;
                }

                if (_tag.HasTag(item, "Plastic"))
                {
                    var junk = Spawn(component.BadRecipeEntityId, Transform(uid).Coordinates);
                    _container.Insert(junk, component.Storage);
                    QueueDel(item);
                    _sharedContainer.EmptyContainer(component.Storage);
                    _audio.PlayPvs(component.FoodDoneSound, uid, AudioParams.Default.WithVolume(-1));
                    return;
                }
```
unsafe items cause it to explode.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 9 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:419]: I feel like it's more realistic that way.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 14 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:574]: yeah, should it not be break though?

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 13 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:460]: apologies, just realized these are literally different things.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 9 above)
Comment (Partmedia) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: Agreed, that's fine.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 14 above)
Comment (Partmedia) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:571]: Adding a `continue` is a pattern that is used elsewhere in the code base and would help soemone reading the code in the future. It's not obvious from one glance but the rest of the code after the time check is being run after every tick (instead of at one second intervals). I think it's important to be consistent with the rest of the code to avoid confusing others.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 18]
```diff
@@ -45,14 +49,22 @@ public sealed class MicrowaveSystem : EntitySystem
...
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+        [Dependency] private readonly ExplosionSystem _explosion = default!;
+        [Dependency] private readonly SharedDestructibleSystem _destruction = default!;
         [Dependency] private readonly SharedContainerSystem _sharedContainer = default!;
         [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
         [Dependency] private readonly TagSystem _tag = default!;
         [Dependency] private readonly TemperatureSystem _temperature = default!;
         [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
         [Dependency] private readonly HandsSystem _handsSystem = default!;
+
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:63]: Extraneous space

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 19]
```diff
@@ -412,20 +452,19 @@ public void Wzhzhzh(EntityUid uid, MicrowaveComponent component, EntityUid? user
...
-                    return;
+                    component.IsSparking = true;
                 }
 
                 if (_tag.HasTag(item, "MicrowaveSelfUnsafe") || _tag.HasTag(item, "Plastic"))
                 {
                     var junk = Spawn(component.BadRecipeEntityId, Transform(uid).Coordinates);
                     _container.Insert(junk, component.Storage);
                     QueueDel(item);
+                    _sharedContainer.EmptyContainer(component.Storage);
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:463]: Was not returning the ingredients considered a bug? To me, it makes sense that if you microwave plastic with your ingredients, that the plastic melts all over the ingredients and wastes them.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 15 above)
Comment (Partmedia) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: I insist that you adjust this code to match our convention.

File: Content.Server/Lightning/LightningSystem.cs [hunk 20]
```diff
@@ -48,15 +49,20 @@ private void OnRemove(EntityUid uid, LightningComponent component, ComponentRemo
     /// <param name="user">Where the lightning fires from</param>
     /// <param name="target">Where the lightning fires to</param>
     /// <param name="lightningPrototype">The prototype for the lightning to be created</param>
-    public void ShootLightning(EntityUid user, EntityUid target, string lightningPrototype = "Lightning")
+    /// <param name="triggerLightningEvents">if the lightnings being fired should trigger lightning events.</param>
+    public void ShootLightning(EntityUid user, EntityUid target, string lightningPrototype = "Lightning",  bool triggerLightningEvents = false)
```
Comment (Partmedia) [inline Content.Server/Lightning/LightningSystem.cs:53]: The old behavior was default true. Please adjust the default to preserve the old behavior. Otherwise, this is a breaking API change and needs to be in a separate PR.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 19 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:463]: if you try microwaving plastic on the master branch it throws an error pretty sure.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 14 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:569]: continue seems to be causing test failures where as break does not.

Comment (Partmedia) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:569]: `break` is a bug, because that means breaking out of the while loop that iterates through all of the microwave components (i.e. exploding one is enough to stop processing the rest of the microwaves). You should investigate the root cause of the test failure, rather than saying "this code is correct because it passes the tests".

Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:572]: Tests seem to be passing now, strange.

Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:572]:
```
if (metaData.EntityPrototype.ID == recipeSolid.Key)
                        {
                            _sharedContainer.Remove(item, component.Storage);
                            EntityManager.DeleteEntity(item);
                            continue;
                        }
```
Removed the break statement here and put a continue, seemed to resolve the test failure?

**Comment by GreyMario** [comment]
you can run tests locally without committing to repo

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 14 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:572]: nvm that wasn't it, lol.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 15 above)
Comment (Tin-Man-Tim) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: sure thing

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 21]
```diff
@@ -45,13 +49,19 @@ public sealed class MicrowaveSystem : EntitySystem
...
+        [Dependency] private readonly ExplosionSystem _explosion = default!;
+        [Dependency] private readonly SharedDestructibleSystem _destruction = default!;
         [Dependency] private readonly SharedContainerSystem _sharedContainer = default!;
         [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
         [Dependency] private readonly TagSystem _tag = default!;
         [Dependency] private readonly TemperatureSystem _temperature = default!;
         [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
         [Dependency] private readonly HandsSystem _handsSystem = default!;
         [Dependency] private readonly SharedItemSystem _item = default!;
+        private TimeSpan _targetTime = TimeSpan.Zero;
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:64]: You should use a floating frameTime instead of a TimeSpan, see `GasAnalyzerSystem.cs` for an example. There is a strong preference for the accumulated frametime to be stored in the component instead of the system.

File: Content.IntegrationTests/Tests/PrototypeSaveTest.cs
Comment (Partmedia) [inline Content.IntegrationTests/Tests/PrototypeSaveTest.cs:1]: This is an unrelated change and shouldn't be in this PR.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 21 above)
Comment (Partmedia) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:64]: The microwave's `CookTimeRemaining` is also a good example of how to do this.

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 22]
```diff
@@ -384,6 +394,32 @@ public static bool HasContents(MicrowaveComponent component)
             return component.Storage.ContainedEntities.Any();
         }
 
+        /// <summary>
+        /// Handles the attempted cooking of unsafe objects
+        /// </summary>
+        /// <remarks>
+        /// Returns true if the microwave can continue to cook, false if it can't.
+        /// </remarks>
+        private bool HandleUnsafeItems(EntityUid uid, MicrowaveComponent component)
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:403]: Since this returns a `bool`, renaming this `RollExplode` or `ShouldExplode` and flipping the boolean makes it more clear what this is doing.

**Comment by Partmedia** [comment]
Thank you for your continued work on this PR. You're almost there.

**Review by AJCM-git — APPROVED** [review]
pushed some reviews, you can see them in the "reviews" commit, just some minor cleanup of things you didnt touch and made your code not use an accumulator but a timespan target
